### PR TITLE
feat(components, protocol-designer): add Thermocycler step details UI

### PR DESCRIPTION
**Pull request by ncdiehl11**
# Overview

Similar to viewing substeps detail for moveLiquid type steps, this PR adds UI for thermocycler substep details. Also, I move the step details toolbox to the right side of the screen as designs specify, and make a few other small styling changes to our ListItem, Toolbox, and StepOverFlowMenu components.

Closes AUTH-882

## Test Plan and Hands on Testing

- create or import a protocol ([example](https://github.com/user-attachments/files/17478058/fullPdWithTC.json.zip)) that contains a Thermocycler profile step
- on the step's overflow menu, select 'View details'
- verify that the step details correctly populate in a toolbox on the right side of the app

https://github.com/user-attachments/assets/ab5d44e9-17dd-42c5-bb67-e64c7766f0cb

## Changelog

- wire up Thermocycler substeps content
- move the substeps toolbox to the right of the screen
- add small style changes to `ListItem` padding and `Toolbox` width/padding
- add tests

## Review requests

- see test plan

## Risk assessment

low

### CODE REVIEW

File: protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/ThermocyclerProfileSubsteps.tsx [hunk 1]
```diff
@@ -1,3 +1,117 @@
...
+                      temperature={temperature}
+                      duration={`${durationMinutes}:${durationSeconds}`}
+                    />
+                  )
+                })}
+                <StyledText
+                  desktopStyle="bodyDefaultRegular"
+                  alignSelf={ALIGN_FLEX_END}
+                >
+                  {`Repeat ${substep.repetitions} times `}
```
Comment (koji) [inline protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/ThermocyclerProfileSubsteps.tsx]: probably adding this to i18next json file would be better.

File: protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/SubstepsToolbox.tsx [hunk 2]
```diff
@@ -47,16 +48,14 @@ export function SubstepsToolbox(
...
-
   return ('commandCreatorFnName' in substeps &&
     (substeps.commandCreatorFnName === 'transfer' ||
       substeps.commandCreatorFnName === 'consolidate' ||
       substeps.commandCreatorFnName === 'distribute' ||
       substeps.commandCreatorFnName === 'mix')) ||
     substeps.substepType === THERMOCYCLER_PROFILE ? (
     <Toolbox
-      width="396px"
+      width={FLEX_MAX_CONTENT}
```
Comment (jerader) [inline protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/SubstepsToolbox.tsx:58]: this works for most cases, but when you have a multi-channel and you're expanding the step, the toolbox width resizes:

https://github.com/user-attachments/assets/2b93f144-c17c-4ba3-bc6d-4e46781ef1c5

**Comment by koji** [comment]
For time format
https://www.figma.com/design/WbkiUyU8VhtKz0JSuIFA45?node-id=11196-717807&m=dev#996135175

File: protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/ThermocyclerProfileSubsteps.tsx (hunk 1 above)
Comment (ncdiehl11) [reply protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/ThermocyclerProfileSubsteps.tsx]: thanks! missed this.

File: protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/SubstepsToolbox.tsx (hunk 2 above)
Comment (ncdiehl11) [reply protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/SubstepsToolbox.tsx:58]: Ah I see, what do you think is the best bet here? The issue with the fixed with is that the toolbox is not always wide enough for the Thermocycler substeps and forces them to wrap

Comment (ncdiehl11) [reply protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/SubstepsToolbox.tsx:58]: Are you comfortable with addressing the toolbox styling in more depth once we add responsiveness?

File: protocol-designer/src/assets/localization/en/protocol_steps.json [hunk 3]
```diff
@@ -105,6 +105,8 @@
       "closed": "closed",
       "open": "open"
     },
+    "repeat": "Repeat {{repetitions}} times",
```
Comment (koji) [inline protocol-designer/src/assets/localization/en/protocol_steps.json:108]: 👍

**Review by koji — APPROVED** [review]
lgtm

File: protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/SubstepsToolbox.tsx (hunk 2 above)
Comment (jerader) [reply protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/SubstepsToolbox.tsx:58]: oooh ya, that sounds good to me!

**Pull request merged**